src/App.js

```diff
@@ -1,34 +1,114 @@
 import React from 'react';
 import './App.css';
 import BackendAdapter from './backend-adapter.js';
-import MetricFieldTypeahead from './metric-field-typeahead.js';
+import AddCharts from './add-charts.js';
+import { default as MetricFieldTypeahead, getMetricNamesFromData } from './metric-field-typeahead.js';
 import LineGraph from './line-graph.js';
+import SingleMetricChart from './single-metric-chart.js';
 import DateTimeRangePicker from '@wojtekmaj/react-datetimerange-picker';
 import 'bootstrap/dist/css/bootstrap.min.css';
 import Button from 'react-bootstrap/Button';
 import ButtonToolbar from 'react-bootstrap/ButtonToolbar';
 
 class App extends React.Component {
 
+  chartTypes = ["count", "max", "avg", "min"]
+
   constructor(props) {
     super(props);
+
+    const chartsJson = window.localStorage.getItem("charts");
+    let charts = [];
+
+    console.log("chartsJson", chartsJson);
+    try {
+      let chartsTemp = JSON.parse(chartsJson);
+      console.log("chartsTemp.length", chartsTemp.length);
+      charts = chartsTemp;
+    } catch (e) {
+      console.error("woa, charts could not be retrieved due to malformed JSON...", chartsJson);
+      console.log("resetting chartsJson");
+      window.localStorage.setItem("charts", "[]");
+    }
+
     this.state = {
       data: [],
       isLoading: true,
-      metricField: ""
+      metricField: "",
+      countData: {},
+      metricNames: [],
+      charts: charts,
+      chartData: new Array(charts.length).fill(0)
     };
   }
 
+  parseQueryParams(paramString) {
+    return paramString.split("&").filter(v => !!v).reduce((params, keyVal) => {
+      const [key, val] = keyVal.split("=");
+      params[key] = val;
+      return params;
+    }, {});
+  }
+
+  getQueryParams() {
+    console.log("query params", window.location.search);
```

Comment:
Does react really not have something built in that wraps query param functionality?

Reply:
React isn't a _framework_ it's a **view** library 😆 

I can get ReactRouter but I didn't wanna do that yet....